editor/src/components/canvas/dom-walker.ts

```diff
@@ -635,57 +637,79 @@ function getGridElementProperties(
     parseGridRange(container, 'column', elementStyle.gridColumn),
   )
 
-  const gridColumnStart =
-    gridColumn?.start ??
-    defaultEither(
-      null,
-      parseGridPosition(
-        container,
-        'column',
-        'start',
-        gridColumn?.start ?? null,
-        elementStyle.gridColumnStart,
-      ),
-    ) ??
-    null
-  const gridColumnEnd =
-    gridColumn?.end ??
-    defaultEither(
-      null,
-      parseGridPosition(
-        container,
-        'column',
-        'end',
-        gridColumn?.end ?? null,
-        elementStyle.gridColumnEnd,
-      ),
-    ) ??
-    null
-  const adjustedColumnEnd =
-    isCSSKeyword(gridColumnEnd) && gridColumn?.end != null ? gridColumn.end : gridColumnEnd
+  const columnSpan: GridSpan | null = isGridSpan(gridColumn?.start)
+    ? gridColumn.start
+    : isGridSpan(gridColumn?.end)
+    ? gridColumn.end
+    : null
+
+  const gridColumnStart = isGridSpan(gridColumn?.start)
+    ? null
+    : gridColumn?.start ??
+      defaultEither(
+        null,
+        parseGridPosition(
+          container,
+          'column',
+          'start',
+          gridColumn?.start ?? null,
+          elementStyle.gridColumnStart,
+        ),
+      ) ??
+      null
+  const gridColumnEnd = isGridSpan(gridColumn?.end)
+    ? null
+    : gridColumn?.end ??
+      defaultEither(
+        null,
+        parseGridPosition(
+          container,
+          'column',
+          'end',
+          gridColumn?.end ?? null,
+          elementStyle.gridColumnEnd,
+        ),
+      ) ??
+      null
+  const adjustedColumnEnd = isGridSpan(gridColumn?.end)
```

Comment:
I'd put these into a helper because these ternary + ?? operator + higher order function combos are a bit hard to parse

Reply:
[`1d4df1b` (#6629)](https://github.com/concrete-utopia/utopia/pull/6629/commits/1d4df1b4b41077fa1af058452e365acc2d933d32)